File: 백준/Bronze/2292. 벌집/벌집.py
# 2292. 벌집

import sys

n = int(sys.stdin.readline())

a, b, cnt, room = 1, 1, 1, 2

while 1:
    if n == 1:
        print(1)
        break

    elif a+1 <= n and n <= 6*b+1:
        print(room)
        break

    room = room + 1
    cnt = cnt + 1  # 2, 3
    a = 6*b+1  # 7, 20
    b = b + cnt  # 3, 6

File: 백준/Bronze/2292. 벌집/벌집.java
import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int n = Integer.parseInt(br.readLine()) - 1;


        if (n == 0) {
            System.out.println(1);
        } else {
            int a = 1;
            int b = 1;
            int cnt = 1;
            int room = 2;

            while (true) {
                if (a <= n && n <= 6 * b) {
                    System.out.println(room);
                    break;
                }

                a = 6 * b + 1;
                cnt++;
                b += cnt;
                room++;
            }
        }


        br.close();
    }
}